hermes-console-vue/src/composables/subscription/use-form-subscription/useFormSubscription.ts

```diff
@@ -0,0 +1,252 @@
+    sendingDelay: [required(), min(0), max(5000)],
+    inflightMessageTTL: [required(), min(0), max(7200)],
+    retryBackoff: [required(), min(0), max(1000000)],
+    retryBackoffMultiplier: [required(), min(1), max(10)],
+    messageDeliveryTrackingMode: [required()],
+    monitoringSeverity: [required()],
+  };
+}
+
+function getRawDataSources(): RawDataSources {
+  const configStore = useAppConfigStore();
+  const allContentTypes = [
+    { title: 'JSON', value: 'JSON', unsupportedDeliveryTypes: [] },
+    {
+      title: 'AVRO (not supported in BATCH delivery mode)',
+      value: 'AVRO',
+      unsupportedDeliveryTypes: ['BATCH'],
+    },
+  ];
+  const deliveryTypes = configStore.loadedConfig.subscription.deliveryTypes.map(
+    (type) => {
+      return {
+        title: type.label,
+        value: type.value,
+      };
+    },
+  );
+  const deliveryModes = [
+    { title: 'ANYCAST', value: 'ANYCAST' },
+    { title: 'BROADCAST (incubating feature)', value: 'BROADCAST' },
+  ];
+  const monitoringSeverities = [
+    { title: 'Non-important', value: 'NON_IMPORTANT' },
+    { title: 'Important', value: 'IMPORTANT' },
+    { title: 'Critical', value: 'CRITICAL' },
+  ];
+  const messageDeliveryTrackingModes = [
+    { title: 'No tracking', value: 'trackingOff' },
+    { title: 'Track message discarding only', value: 'discardedOnly' },
+    { title: 'Track everything', value: 'trackingAll' },
+  ];
+  const fetchedOwnerSources = ref<OwnerSource[]>([]);
+  const ownerSources = computed(() =>
+    fetchedOwnerSources.value
+      .filter((source) => !source.deprecated)
+      .map((source) => {
+        return { title: source.name, value: source };
+      }),
+  );
+  fetchOwnersSources().then(
+    (response) => (fetchedOwnerSources.value = response.data),
+  );
+  const owners = ref<SelectFieldOption[]>([]);
+  const loadingOwners = ref(false);
+
+  return {
+    allContentTypes,
+    deliveryTypes,
+    deliveryModes,
+    monitoringSeverities,
+    messageDeliveryTrackingModes,
+    fetchedOwnerSources,
+    ownerSources,
+    owners,
+    loadingOwners,
+  };
+}
+
+function createEmptyForm(): Ref<SubscriptionForm> {
+  return ref({
+    name: '',
+    endpoint: '',
+    description: '',
+    ownerSource: null,
+    owner: '',
+    ownerSearch: '',
+    contentType: '',
+    deliveryType: '',
+    mode: '',
+    subscriptionPolicy: {
+      rateLimit: null,
+      inflightMessageTTL: 3600,
+      retryBackoff: 1000,
+      retryBackoffMultiplier: 1.0,
+      sendingDelay: 0,
+      requestTimeout: 1000,
+      batchSize: null,
+      batchTime: null,
+      batchVolume: null,
+    },
+    retryOn4xx: false,
+    messageDeliveryTrackingMode: '',
+    monitoringDetails: {
+      severity: '',
+      reaction: '',
+    },
+    deliverUsingHttp2: false,
+    attachSubscriptionIdentityHeaders: false,
+    deleteSubscriptionAutomatically: false,
+    pathFilters: [],
+    headerFilters: [],
+  });
+}
+
+export function initializeFulfilledForm(
```

Comment:
suggestion: `Fulfilled` is not the right word to use in this context. Use `FullyFilled` instead